src/System.Data.SqlClient/src/System/Data/SqlClient/TdsParserStateObjectNative.cs

```diff
@@ -0,0 +1,363 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Data.Common;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using System.Threading.Tasks;
+
+namespace System.Data.SqlClient
+{
+    internal class TdsParserStateObjectNative : TdsParserStateObject
+    {
+        private SNIHandle _sessionHandle = null;              // the SNI handle we're to work on
+
+        private SNIPacket _sniPacket = null;                // Will have to re-vamp this for MARS
```

Comment:
Do we still need this comment at the end of this line? Not sure what it's referring to.